fmf/_compat_jsonschema.py

```diff
@@ -0,0 +1,91 @@
+"""Compatibility layer for jsonschema validation."""
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Dict, List, NamedTuple, Optional, Union
+
+import jsonschema
+
+try:
+    from jsonschema import validators
+    from referencing import Registry, create
+    from referencing.jsonschema import DRAFT4
+    MODERN_JSONSCHEMA = True
+except ImportError:
+    MODERN_JSONSCHEMA = False
+
+
+class JsonSchemaValidationResult(NamedTuple):
+    """Represents JSON Schema validation result."""
+    result: bool
+    errors: List[Any]
+
+
+def create_validator(
+        schema: Dict[str, Any],
+        schema_store: Optional[Dict[str, Any]] = None
+        ) -> Union[jsonschema.validators.Validator, jsonschema.validators.Draft4Validator]:
+    """Create a validator instance based on available jsonschema version."""
+    schema_store = schema_store or {}
+
+    if MODERN_JSONSCHEMA:
+        # Modern approach with referencing
+        registry: Registry = Registry().with_resources([
+            (uri, create({'$schema': 'http://json-schema.org/draft-04/schema#', **contents}))
+            for uri, contents in schema_store.items()
+            ])
+        cls = validators.create(meta_schema=DRAFT4, validators={})
+        return cls(schema=schema, registry=registry)
+    else:
+        # Legacy approach with RefResolver
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=DeprecationWarning)
```

Comment:
Let's try without the `catch_warnings`. There should not be the DeprecationWarning being triggered if the `MODERN_JSONSCHEMA` is working appropriately, and if it does, we should investigate why it does.

Reply:
True. Removed.